### PR TITLE
Tools 102 easier added fields

### PR DESCRIPTION
**Pull request by jychien**
This version of the flattener corresponds with [this comment](https://lattice.atlassian.net/browse/TOOLS-102?focusedCommentId=15857) in TOOL-102, with the following additions:

- Fix for cell identifier logging error (where RawMatrixFiles without associate prefix/suffix, such as Slide-seq) are taken into consideration
- HumanPostnatalDonor.living_at_sample_collection (donor_living_at_sample_collection)
- HumanPostnatalDonor.menopausal_status (donor_menapausal_status)
- HumanPostnatalDonor.smoker (donor_smoking_status)
- Tissue.derivation_process (sample_derivation_process)
- Suspension.dissociation_reagent (suspension_dissociation_reagent)
- Suspension.dissociation_time/Suspension.dissociation_time_units (suspension_dissociation_time)
- Suspension.depleted_cell_types (suspension_depleted_cell_types)
- Suspension.derivation_process (suspension_derivation_process)
- Suspension.percent_cell_viability (suspension_percent_cell_viability)
- Suspension.tissue_handling_interval (tissue_handling_interval)
- Library.starting_quantity/Library.starting_quantity_units (library_starting_quantity)
- SequencingRun.platform
- RawMatrixFile.assembly
- RawMatrixFile.software
- Tissue.source

Related tickets that have recently gotten in that should also be taken into consideration:
- Convert bmi to numeric: https://lattice.atlassian.net/browse/TOOLS-124
- Cell identifiers and annotation logging: https://lattice.atlassian.net/browse/TOOLS-125

Would be great to have a thorough code review and testing. @jahilton @Jchaffer787, not sure what people's bandwidth are or best plan of action. I can update this ticket later with a list of ProcMatrixFiles to run? Should I start up EC2?

### CODE REVIEW

File: scripts/flattener.py [hunk 1]
```diff
@@ -151,7 +180,7 @@
 
 EPILOG = '''
 Examples:
-
+HsapDv:0000264
```
Comment (jahilton) [inline scripts/flattener.py:183]: I'm not understanding why this is here

Comment (jychien) [reply scripts/flattener.py:183]: Oh dear, that should not be there. Cut and paste error. I'll get all the fixes in once you've had a chance to review the remainder of the code.

**Review by jahilton — APPROVED** [review]

**Comment by jahilton** [comment]
This code looks good. I would suggest running various matrix files through it for validation before merging

**Comment by jychien** [comment]
Files to flatten:
- living_at_sample_collection:
     - **LATDF578KWM**: donors are living_at_sample_collection = False
     - **LATDF742BQI**: donors are living_at_sample_collection = True
     - **LATDF273MDG**: donors are living_at_sample_collection are mixed
- menopausal_status: **LATDF964MKP**
- donor_smoking_status: **LATDF773LWI** (Ye lupus, very large)
- sample_derivation_process: review on other test datasets to make sure it is accurate
- suspension_dissociation_reagent: review on other test datasets to make sure it is accurate
- suspension_dissociation_time: **LATDF519CUZ**
- suspension_depleted_cell_types: **LATDF925LET**
- suspension_derivation_process:  review on other test datasets to make sure it is accurate
- suspension_percent_cell_viability: **LATDF287ORI**
- tissue_handling_interval: **LATDF700WAP**
- library_starting_quantity: **LATDF448PLU**
- Tissue.source: **LATDF273MDG**
- SequencingRun.platform: **LATDF013ZKP**
- RawMatrixFile.annotation: **LATDF093IXU**
- RawMatrixFile.assembly: **LATDF726UIF**
- RawMatrixFile.software: **LATDF909YSO**
- Unique assay regression: 
     - Slide-seq: **LATDF329OGL**
     - ATAC: **LATDF180GFI**
     - Visium: **LATDF584NGT**

**Comment by jychien** [comment]
**Lian**:
- LATDF578KWM
- LATDF742BQI
- LATDF273MDG
- LATDF964MKP
- LATDF519CUZ
- LATDF925LET

**Jim**:
- LATDF287ORI
- LATDF700WAP
- LATDF448PLU
- LATDF273MDG
- LATDF013ZKP
- LATDF093IXU

**Jenny**:
- LATDF726UIF
- LATDF909YSO
- LATDF329OGL
- LATDF180GFI
- LATDF584NGT
- LATDF483VSD

**Together**: LATDF773LWI 

**Things to do/look at:**
- Run validator on h5ad
- Look at dataset currently in cellxgene portal and look at old fields and make sure new fields makes sense
- For the specific dataset, use jupyter notebook to make sure the associated field has values that make sense. Things to watch out for are 'unknown' values, and if we may need to change that to a different default value. For example, `suspension_depleted_cell_types` should default to 'na' rather than 'unknown'.
- Check variable types, as `suspension_percent_cell_viability` and `donor_BMI_at_collection` should be all numeric if there are not any '>', '<' or '-' in the string
- If extra time allows, would be great to have each of us pick another one to run, as I may have been unknowingly biased in choosing these ProcMatrixFiles.

**Pull request merged**